### ISSUE TITLE
Difference in decoding of OCGs names between `doc.get_ocgs()` and `page.get_drawings()`

### ISSUE THREAD

**Issue by Rodrigodd** [pymupdf/PyMuPDF#3143]
### Description of the bug

The list returned by `doc.get_ocgs()`  contains the expected names of the OCGs, but the ones in the `'layer'` field of the elements in `page.get_drawings()` are misdecoded.

### How to reproduce the bug

Running the following code:

```python
import fitz
import sys
from pprint import pprint

filename = sys.argv[1]
doc = fitz.open(filename)

pprint(doc.get_ocgs())

for page in doc:
    print('# Page:', page.number)
    print('## bboxlog layers')
    for _, _, l in page.get_bboxlog(layers=True):
        print(l)
    print('## drawing layers')
    for obj in page.get_drawings():
        print(obj.get('layer'))
```

With the following file:

[layers-test2.pdf](https://github.com/pymupdf/PyMuPDF/files/14212440/layers-test2.pdf)

Gives the following output:

```shell
PS > python .\show_layers.py .\layers-test2.pdf
{1: {'intent': [], 'name': '0', 'on': True, 'usage': None},
 2: {'intent': [], 'name': 'braço', 'on': True, 'usage': None},
 3: {'intent': [], 'name': 'braço1', 'on': True, 'usage': None},
 4: {'intent': [], 'name': 'ênvelo', 'on': True, 'usage': None},
 5: {'intent': [], 'name': 'camada', 'on': True, 'usage': None}}
# Page: 0
## bboxlog layers
0
braÃ§o
braÃ§o
braÃ§o1
Ãªnvelo
camada
## drawing layers
0
braÃ§o
braÃ§o
braÃ§o1
Ãªnvelo
camada
```

### PyMuPDF version

1.23.21

### Operating system

Windows

### Python version

3.12

**Comment by JorjMcKie** [comment]
Thanks for submitting this.
We will look at it now.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.22.
